tests/integration/SdkTest_test.cpp

```diff
@@ -4839,7 +4839,7 @@ TEST_F(SdkTest, SdkBackupFolder)
     }
     else
     {
-        auto now = m_time(nullptr);
+        const time_t now = m_time(nullptr);
         char timebuf[32];
         strftime(timebuf, sizeof timebuf, "%c", localtime(&now));
 
```

Comment:
It turns out time_t is a bit inconsistent across platforms and 32/64 bit compilers.  We have smoothed that out with m_time and some other related functions all beginning with m_, which operate on m_time_t instead.  I think the right solution here is to call m_localtime instead?    Also, do we need that 'const' ?  In C++ the default it that variables are variable - I think we should go with the C++ flow, rather than fight the language, and only worry about const where we want to remind other devs that something would break if they decided to actually modify it.  thanks

Reply:
I looked at the header with `m_localtime` defined (and other uses in the SDK) and I think I've replaced this to use it correctly, let me know.

The issue of `const` is now sidestepped entirely (if I've used the functions correctly, was following [this](https://github.com/meganz/sdk/blob/13df5bc543ed21b1edd15010593790ae095c1bed/src/sync.cpp#L2005))but my personal conventions (and justifications) lie here: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#con1-by-default-make-objects-immutable 
I think a lot of C++ defaults are inherited for C compatibility or arguably dated conventions, and we should evaluate them for safety/clarity/performance wherever possible--as a developer glancing at someone else's code it's much easier to see `const` and know the variable cannot change (and that I will get a compiler error if I try to do so) rather than see a mutable variable and have to mentally keep in mind that it might change somewhere

Reply:
thanks for the fix, and the explanation - I've sent you a counterargument for that one - the article is advice for beginners really, and doesn't take into account whether you're in a complicated situation where it matters, or a simple one where it's just redundant and distracting extra syntax.